### PR TITLE
[voq] Add fabric counter cli commands

### PR DESCRIPTION
**Pull request by mlorrillere**
Add fabric counters CLI commands.

Commands are:
show fabric counters port
show fabric counters port -n asic4
show fabric counters port -n asic5 -e
show fabric counters queue
show fabric counters queue -n asic4

Signed-off-by: Maxime Lorrillere <mlorrillere@arista.com>

#### New command output (if the output of a command-line utility has changed)

```console
# fabricstat -h
usage: fabricstat [-h] [-q] [-n NAMESPACE] [-e]

Display the fabric port state and counters

optional arguments:
  -h, --help            show this help message and exit
  -q, --queue           Display fabric queue stat, otherwise port stat
  -n NAMESPACE, --namespace NAMESPACE
                        Display fabric ports for specific namespace
  -e, --errors          Display errors

Examples:
    fabricstat
    fabricstat --namespace asic0
    fabricstat -p -n asic0 -e
    fabricstat -q
    fabricstat -q -n asic0
```

```console
# show fabric counters -h
Usage: show fabric counters [OPTIONS] COMMAND [ARGS]...

  Show fabric port counters

Options:
  -h, -?, --help  Show this message and exit.

Commands:
  port   Show fabric port stat
  queue  Show fabric queue stat
```

```console
# show fabric counters port -h
Usage: show fabric counters port [OPTIONS]

  Show fabric port stat

Options:
  -d, --display [all|frontend]    Show internal interfaces  [default:
                                  frontend]
  -n, --namespace [asic0|asic1|asic2|asic3|asic4|asic5|asic6|asic7|asic8|asic9|asic10|asic11]
                                  Namespace name or all
  -e, --errors
  -h, -?, --help                  Show this message and exit.
```

Example output:
```console
# show fabric counters port -n asic0
  ASIC    PORT    STATE    IN_CELL    IN_OCTET    OUT_CELL    OUT_OCTET    CRC    FEC_CORRECTABLE    FEC_UNCORRECTABLE    SYMBOL_ERR
------  ------  -------  ---------  ----------  ----------  -----------  -----  -----------------  -------------------  ------------
     0       0     down          0           0           0            0      0             206318           7949306448             0
     0       1     down          0           0           0            0      0                  0           7949048925             0
     0       2     down          0           0           0            0      0                  0           7948552578             0
     0       3     down          0           0           0            0      0                  0           7948049934             0
     0       4     down          0           0           0            0      0                  0           7947524845             0
     0       5     down          0           0           0            0      0                  0           7946976870             0
     0       6     down          0           0           0            0      0                  0           7946408318             0
     0       7     down          0           0           0            0      0                  0           7945775522             0
[...]
     0     187     down          0           0           0            0   3024         6887370202              3773897   11545823075
     0     188     down          0           0           0            0      0                  0          13130704211             0
     0     189     down          0           0           0            0      0             118941          13129730022             0
     0     190       up          0           0           0            0    514          162556197           2717957083          4823
     0     191     down          0           0           0            0      0                  0          13128089835             0
```

### CODE REVIEW

**Comment by rlhui** [comment]
@skeesara-nokia, @vganesan-nokia - please help review, thanks

File: scripts/fabricstat [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+from utilities_common import constants
+from natsort import natsorted
+from tabulate import tabulate
+import utilities_common.multi_asic as multi_asic_util
+
+ASIC_NAME_PREFIX = 'asic'
+PORT_NAME_PREFIX = 'PORT'
+COUNTER_TABLE_PREFIX = "COUNTERS:"
+FABRIC_PORT_STATUS_TABLE_PREFIX = "FABRIC_PORT_TABLE|"
+FABRIC_PORT_STATUS_FIELD = "STATUS"
```
Comment (arlakshm) [inline scripts/fabricstat:15]: Can we add some unit tests for this?

File: scripts/fabricstat [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        return cnstat_dict
+
+    def cnstat_print(self, cnstat_dict, errors_only=False):
+        if len(cnstat_dict) == 0:
+            print("Counters %s empty" % self.namespace)
+            return
+
+        table = []
+        header = None
+        asic = int(self.namespace[len(ASIC_NAME_PREFIX):])
```
Comment (arlakshm) [inline scripts/fabricstat:109]: We can use this function defined in sonic_py_common
[`get_asic_id_from_name(asic_name):` ](https://github.com/Azure/sonic-buildimage/blob/935923b8059a48b01a4daa765cdfa4bb6560562a/src/sonic-py-common/sonic_py_common/multi_asic.py#L136)

**Review by arlakshm — CHANGES_REQUESTED** [review]

File: scripts/fabricstat (hunk 1 above)
Comment (mlorrillere) [reply scripts/fabricstat:21]: Added unit test.

**Comment by arlakshm** [comment]
> This pull request **introduces 9 alerts** when merging [c8cc5c7](https://github.com/Azure/sonic-utilities/commit/c8cc5c78c9aaef24e7fe087894e99ed04937a59a) into [c05845d](https://github.com/Azure/sonic-utilities/commit/c05845df1567c6d14312bb36f35c72c7ea6a93ee) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-utilities/rev/pr-9e3a97535f012ecb305c9d18804a0043304d989c)
> 
> **new alerts:**
> 
> * 8 for Unused import
> * 1 for Unused local variable

@mlorrillere can you fix the lgtm alerts?

**Comment by arlakshm** [comment]
> This pull request **introduces 9 alerts** when merging [c8cc5c7](https://github.com/Azure/sonic-utilities/commit/c8cc5c78c9aaef24e7fe087894e99ed04937a59a) into [c05845d](https://github.com/Azure/sonic-utilities/commit/c05845df1567c6d14312bb36f35c72c7ea6a93ee) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-utilities/rev/pr-9e3a97535f012ecb305c9d18804a0043304d989c)
> 
> **new alerts:**
> 
> * 8 for Unused import
> * 1 for Unused local variable

@mlorrillere can you fix the lgtm alerts?

**Comment by mlorrillere** [comment]
> > This pull request **introduces 9 alerts** when merging [c8cc5c7](https://github.com/Azure/sonic-utilities/commit/c8cc5c78c9aaef24e7fe087894e99ed04937a59a) into [c05845d](https://github.com/Azure/sonic-utilities/commit/c05845df1567c6d14312bb36f35c72c7ea6a93ee) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-utilities/rev/pr-9e3a97535f012ecb305c9d18804a0043304d989c)
> > **new alerts:**
> > 
> > * 8 for Unused import
> > * 1 for Unused local variable
> 
> @mlorrillere can you fix the lgtm alerts?

My bad, I had the changes ready and forgot to commit. Should be fixed now.

**Comment by arlakshm** [comment]
Hi @mlorrillere, the unit tests added for the fabric counter, `tests/fabricstat_test.py`,  are failing.
I think the reason they are failing is because the tests are trying to connect to the actual redis instance iso for mock db.
the script `fabricstat` the mock db must if the `UTILITIES_UNIT_TESTING` enivornment variable is set.

Please refer to this block as an example https://github.com/Azure/sonic-utilities/blob/master/scripts/intfutil#L15-L28

**Comment by mlorrillere** [comment]
Hi @arlakshm 

I've addressed the remaining issues and all checks passed.

**Review by arlakshm — APPROVED** [review]

**Comment by kenneth-arista** [comment]
@arlakshm since everything is approved, can you merge this PR?

**Pull request merged**